code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -836,24 +838,12 @@ func (sa *StorageAllocation) removeBlobber(
 	}
 	delete(sa.BlobberMap, removeId)
 
-	found = false
-	for i, d := range sa.Blobbers {
-		if d.ID == removeId {
-			sa.Blobbers[i] = sa.Blobbers[len(sa.Blobbers)-1]
-			sa.Blobbers = sa.Blobbers[:len(sa.Blobbers)-1]
-			found = true
-			break
-		}
-	}
-	if !found {
-		return nil, fmt.Errorf("cannot find blobber %s in allocation", remove.BlobberID)
-	}
 	var removedBlobber *StorageNode
 	found = false
 	for i, d := range blobbers {
 		if d.ID == removeId {
 			removedBlobber = blobbers[i]
-			blobbers[i] = blobbers[len(sa.Blobbers)-1]
+			blobbers[i] = blobbers[len(sa.BlobberDetails)-1]
```

Comment:
Incorrect assignment, the previous code was correct.

Reply:
why is this wrong @peterlimg ?
If you track back this function call, it is reading from blobberDetails and removing from there..

Reply:
ahh got it!
yes makes sense..

I'll change this in any of my next PR